ft/cachetable/cachetable.h

```diff
@@ -108,7 +108,14 @@ uint32_t toku_get_cleaner_iterations_unlocked (CACHETABLE ct);
 // create and initialize a cache table
 // size_limit is the upper limit on the size of the size of the values in the table
 // pass 0 if you want the default
-int toku_cachetable_create(CACHETABLE *result, long size_limit, LSN initial_lsn, struct tokulogger *logger);
+int toku_cachetable_create2(CACHETABLE *result, long size_limit,
+                           unsigned long client_pool_threads,
+                           unsigned long cachetable_pool_threads,
+                           unsigned long checkpoint_pool_threads,
+                           LSN initial_lsn, struct tokulogger *logger);
+
+#define toku_cachetable_create(r, s, l, o) \
+    toku_cachetable_create2(r, s, 0, 0, 0, l, 0);
```

Comment:
Yes, I started out that way, well, without the default values and found that hundreds of test cases needed changing. I'm not particularly fond of mixing C++ default values with straight C type API calls, but I do get your point. This was an old habit of sorts from InnoDB and others.

In reality I would much rather a multi-phased creation-initialization-start approach that would be something like:
`toku_cachetable_create(...);
toku_cachetable_set_something(...);
toku_cachetable_set_something(...);
toku_cachetable_set_something(...);
toku_cachetable_start(...);`